bundles/org.openhab.binding.globalcache/src/main/java/org/openhab/binding/globalcache/internal/handler/GlobalCacheHandler.java

```diff
@@ -554,14 +547,16 @@ private void writeCommandToDevice(RequestMessage requestMessage) throws IOExcept
          */
         private String readReplyFromDevice(RequestMessage requestMessage) throws IOException {
             logger.trace("Processor for thing {} reading reply from device", thingID());
-
             if (connectionManager.getCommandIn() == null) {
                 logger.debug("Error reading from device because input stream object is null");
                 return "ERROR: BufferedReader is null!";
             }
-
-            logger.trace("Processor for thing {} reading response from device", thingID());
-            return connectionManager.getCommandIn().readLine().trim();
+            String reply = connectionManager.getCommandIn().readLine();
+            if (reply == null) {
+                logger.debug("Read of reply from device returned null!");
+                return "ERROR: reply is null!";
```

Comment:
It seems that it would be more appropriate to throw `IOException` here, so the error would be handled correctly. Now the method returns successfully with reply "ERROR: reply is null!", leaving the caller no options for handling the situation gracefully.

Reply:
While I agree it would make sense to do that, I was trying to follow the model that I used way back when I initially wrote the binding (it was the first binding that I wrote and I definitely would do some things differently knowing what I know now about openHAB and Java).

When the GlobalCache devices have an error, they reply with a message of the form "ERROR: some descriptive message". So I was trying to be consistent with that model. In fact if you look directly above at line 560, that's exactly how a null input stream object is handled.

Reply:
OK, I didn't have this knowledge that the prefix "ERROR: " would be recognized and handled somewhere else. So in this case the fix is following the convention, however obscure that is. :)